### PR TITLE
General code cleanup: eliminate most preprocessor directives, minor other cleanups

### PR DESCRIPTION
**Pull request by nhaehnle**
The major aim of this series is to make the LLPC code base more accessible by removing the bulk of preprocessor directives. Those directives largely serve no purpose:

We will never want to build with only a subset of hardware support, and even if we wanted to, those subset builds are likely broken most of the time because we don't test them.

The interface version directives can serve a purpose by allowing us to more flexibly combine different version of LLPC and the client driver (this can be rather useful e.g. for bisecting problems). That said:

1. It makes sense to remove backwards compatibility after some time has passed -- as long as LLPC is not more widely deployed, half a year seems like a reasonable cutoff.
2. We should generally strive to do interface changes in a way that does not require preprocessor directives at all; e.g., when adding new fields to structs that are zero-initialized, preprocessor directives are usually unnecessary.

My apologies for packing multiple commits into a single pull request. The reality of it is that those commits are somewhat interdependent, and this seems to be the only way to make those dependencies clear.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

File: llpc/include/llpc.h [hunk 1]
```diff
@@ -45,6 +45,18 @@
 /// LLPC minor interface version.
 #define LLPC_INTERFACE_MINOR_VERSION 1
 
+#ifndef LLPC_CLIENT_INTERFACE_MAJOR_VERSION
+#if VFX_INSIDE_SPVGEN
+#define LLPC_CLIENT_INTERFACE_MAJOR_VERSION LLPC_INTERFACE_MAJOR_VERSION
```
Comment (Flakebi) [inline llpc/include/llpc.h]: Should spvgen define `LLPC_CLIENT_INTERFACE_MAJOR_VERSION` instead?

**Comment by amdrexu** [comment]
Should rebase the commits.

**Comment by JaxLinAMD** [comment]
please reslove conflicts

**Comment by nhaehnle** [comment]
Rebased now

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3e2128b26ae6ddbb39e7d9ed1f4f5219c2abfc6b
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 9fab16015e522fff05890a045a1e9d8d3c23a636</li>
<li>SPVGEN: dev a61db14f75728e91c243d4bfc1c39700e9dc0ba6</li>
<li>XGL: dev 052ebfa2e77abd14fdb4f5120b35b081f3fc4bdc</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 1ecf0a1496f32eaf34825b3d11432c6b1d03c951</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 942/942 (100.0%)</li>
<li>Failed: 0/942 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 65463/65463 (100.0%)</li>
<li>Failed: 0/65463 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by JaxLinAMD** [comment]
please resolve conflicts

**Comment by nhaehnle** [comment]
retest this please

**Comment by nhaehnle** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 46b4803a4f536c4c86a79e6d2d38db88a042b38b
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 9fab16015e522fff05890a045a1e9d8d3c23a636</li>
<li>SPVGEN: dev a61db14f75728e91c243d4bfc1c39700e9dc0ba6</li>
<li>XGL: dev 052ebfa2e77abd14fdb4f5120b35b081f3fc4bdc</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 1ecf0a1496f32eaf34825b3d11432c6b1d03c951</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 942/942 (100.0%)</li>
<li>Failed: 0/942 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 65463/65463 (100.0%)</li>
<li>Failed: 0/65463 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 2154/2155 (100.0%)</li>
<li><details><summary>Failed: 1/2155 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.object_management.multithreaded_shared_resources.graphics_pipeline
Stack trace: result: VK_ERROR_OUT_OF_HOST_MEMORY at vktApiObjectManagementTests.cpp:1814
```
</p></details></li>


</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 65431/65431 (100.0%)</li>
<li>Failed: 0/65431 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by nhaehnle** [comment]
retest this please

**Comment by nhaehnle** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit d5a688a09ce45722ab20742fe008be1e318ee422
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 9fab16015e522fff05890a045a1e9d8d3c23a636</li>
<li>SPVGEN: dev a61db14f75728e91c243d4bfc1c39700e9dc0ba6</li>
<li>XGL: dev 052ebfa2e77abd14fdb4f5120b35b081f3fc4bdc</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 1ecf0a1496f32eaf34825b3d11432c6b1d03c951</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 942/942 (100.0%)</li>
<li>Failed: 0/942 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 65463/65463 (100.0%)</li>
<li>Failed: 0/65463 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 65463/65463 (100.0%)</li>
<li>Failed: 0/65463 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 65431/65431 (100.0%)</li>
<li>Failed: 0/65431 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**